intermediate_source/inductor_debug_cpu.py

```diff
@@ -0,0 +1,622 @@
+
+def neg(x):
+    return f"decltype({x})(-{x})"
+
+######################################################################
+# In order to demonstrate the debugging, we will modify the function to a wrong one later.
+#
+#
+# Errors debugging
+# ^^^^^^^^^^^^^^^^
+#
+# If a compile error occurs, the root cause is usually shown in the traceback log.
+#
+# For example, the ``neg`` function is modified like this:
+
+def neg(x):
+    return f"-{x}"
+
+
+######################################################################
+# The logging gives the following compile error with a rather clear reason. In this case, the root cause is that data types of maximum's inputs are inconsistent.
+#
+# .. code:: shell
+#
+# 	…
+# 	torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
+# 	CppCompileError: C++ compile error
+# 	…
+# 	/tmp/torchinductor_root/2x/c2xgxsooklulr4u54etfnnha7dsu6xzbwdscttvs7dkpba3uwkem.cpp: In function ‘void kernel(const unsigned char*, const unsigned char*, unsigned char*)’:
+# 	/tmp/torchinductor_root/2x/c2xgxsooklulr4u54etfnnha7dsu6xzbwdscttvs7dkpba3uwkem.cpp:14:53: error: no matching function for call to ‘max_propagate_nan(unsigned char&, int&)’
+# 	   14 |             auto tmp3 = max_propagate_nan(tmp0, tmp2);
+# 	      |                                                     ^
+# 	In file included from /tmp/torchinductor_root/2x/c2xgxsooklulr4u54etfnnha7dsu6xzbwdscttvs7dkpba3uwkem.cpp:2:
+# 	/tmp/torchinductor_root/gv/cgv6n5aotqjo5w4vknjibhengeycuattfto532hkxpozszcgxr3x.h:27:17: note: candidate: ‘template<class scalar_t> scalar_t max_propagate_nan(scalar_t, scalar_t)’
+# 	   27 | inline scalar_t max_propagate_nan(scalar_t a, scalar_t b) {
+# 	      |                 ^~~~~~~~~~~~~~~~~
+# 	/tmp/torchinductor_root/gv/cgv6n5aotqjo5w4vknjibhengeycuattfto532hkxpozszcgxr3x.h:27:17: note:   template argument deduction/substitution failed:
+# 	/tmp/torchinductor_root/2x/c2xgxsooklulr4u54etfnnha7dsu6xzbwdscttvs7dkpba3uwkem.cpp:14:53: note:   deduced conflicting types for parameter ‘scalar_t’ (‘unsigned char’ and ‘int’)
+# 	   14 |             auto tmp3 = max_propagate_nan(tmp0, tmp2);
+# 	      |                                                     ^
+#
+#
+# Otherwise, if the model runs with other errors, we can do the model code reduction until finding the minimum code snippet with failure. Thus, the target operators and kernels are located.
+#
+#
+# Accuracy debugging
+# ^^^^^^^^^^^^^^^^^^^
+#
+# The accuracy problem refers the case where outputs of backends eager and inductor are different. As FX graph is generated before Inductor and output code is generated after Inductor, we can narrow down the problem by comparing their outputs.
+#
+# If a model has several graphs, the first step is to compare the final outputs of FX graph and output code for each graph, given the same input. The target is to find the first graph occurring error or with different outputs. Binary search is suggested to use for efficiency.
```

Comment:
`The target is to find the first graph occurring error or with different outputs`

This sentence seems confusing to me. Since the issue is related to accuracy, what does `first graph occurring error` mean here? Thanks